### PR TITLE
feat: Add multiple window support

### PR DESCRIPTION
**Pull request by erdkse**
This PR...

## Changes

- Adds multiple windows feature

## Fixes

-

## How to test it

- Right click on the dock menu and click 'New Window' option

## Screenshots

-

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

**Comment by catalinhoha** [comment]
Let's add the following type for the redux store in `main-store.ts`:
```typescript
import {AnyAction, EnhancedStore} from '@reduxjs/toolkit';
import {ThunkMiddleware} from 'redux-thunk';

export type StoreType = EnhancedStore<RootState, AnyAction, [ThunkMiddleware<RootState, AnyAction>]>;
```
Then, use that type on all occurrences of `store: any` so we can keep TS support. 

---

and I think you have a small typo: `fluxStandartAction` instead of `fluxStandardAction`

**Comment by erdkse** [comment]
@catalinhoha done!

**Comment by olensmar** [comment]
@erdkse does this manage one store per window so there are no conflicts? what happens when settings are updated in one window - are they applied to all windows or only to any newly opened ones?

**Comment by erdkse** [comment]
> @erdkse does this manage one store per window so there are no conflicts? what happens when settings are updated in one window - are they applied to all windows or only to any newly opened ones?

All windows stores are different, they work individually and they are managed by the main renderer's store. My implementation covers 90% of store communications. But still, we need to make some refactoring for communication on renderer stores => main store, main store => renderer stores.

**Comment by erdkse** [comment]
Because all windows read the same electron-store values These kinds of changes will be applied to all windows. We may want to do a refactoring for electron store too. i.e If we set `Automatically load last folder` to true. It will be set to all windows

**Comment by olensmar** [comment]
two requests: 
- can we add a "New Window" option to the File menu? 
- can we show the folder / cluster in the window title?

**Comment by erdkse** [comment]
> two requests:
> 
> * can we add a "New Window" option to the File menu?
> * can we show the folder / cluster in the window title?

yes of course

**Comment by erdkse** [comment]
Ready to review @olensmar

File: electron/menu.ts [hunk 1]
```diff
@@ -47,12 +47,19 @@ const appMenu = (win: BrowserWindow, store: any): MenuItemConstructorOptions =>
...
 
-const fileMenu = (win: BrowserWindow, store: any): MenuItemConstructorOptions => {
+const fileMenu = (store: any): MenuItemConstructorOptions => {
   const configState: AppConfig = store.getState().config;
   const mainState: AppState = store.getState().main;
   return {
     label: 'File',
     submenu: [
+      {
+        label: 'New Window',
```
Comment (olensmar) [inline electron/menu.ts:57]: "New Monokle Window" ?

Comment (olensmar) [reply electron/menu.ts:57]: Do we need a corresponding "Close Window" in the Window menu?

Comment (erdkse) [reply electron/menu.ts:57]: Hmm I can add it. No big deal

File: README.md [hunk 2]
```diff
@@ -50,9 +50,20 @@ simply click `Move to Bin`. No wait, did you? You should have clicked `Open`, si
 
 ## Building / running
 
-### Mac
+Clone this repo, make sure you are running the node version specified in `.npmrc`, then run:
+
+cp .env.example.ts .env.ts
```
Comment (olensmar) [inline README.md:55]: put this in a code block

File: README.md [hunk 3]
```diff
@@ -50,9 +50,20 @@ simply click `Move to Bin`. No wait, did you? You should have clicked `Open`, si
...
+
+to include env variables.
+
+<details>
+  <summary>
+  Note: Default env is set to `production`. You need to change it to `development` for development usage.
```
Comment (olensmar) [inline README.md:61]: just indenting with '>' should enough? (no details/summary tags needed)

**Comment by erdkse** [comment]
Done @olensmar

**Review by olensmar — APPROVED** [review]

**Pull request merged**